tools/integration_tests/cloud_profiler/with_gcp_profiler_service_test.go

```diff
@@ -0,0 +1,203 @@
+	}
+	testDirPath := path.Join(setup.MntDir(), testDirName)
+	filePath := path.Join(testDirPath, "a.txt")
+	setup.SetupTestDirectory(testDirName)
+	operations.CreateFileWithContent(filePath, 0644, string(randomData), t)
+	// Start Workload and Allow Time for Profiling
+	performRepeatedRead(t.Context(), t, filePath, 3*time.Minute)
+	time.Sleep(time.Minute)
+
+	// 1. Fetch GCP projectID.
+	// 2. Create a profiler service api client.
+	// 3. Make list call to the profiler service api client and fetch the profiles.
+	// 4. Filter and match if the right profile data exists.
+	fetchProjectCtx, fetchProjectCancel := context.WithTimeout(context.Background(), 5*time.Second)
+	defer fetchProjectCancel()
+	projectID, err := metadata.ProjectIDWithContext(fetchProjectCtx) // Reduced timeout, 5s is usually sufficient.
+	if err != nil {
+		t.Logf("metadata.ProjectIDWithContext failed: %v, try fetching from environment variable.", err)
+		projectID := os.Getenv("GOOGLE_CLOUD_PROJECT")
+		if projectID == "" {
+			t.Skip("Not able to fetch project ID from metadata server or GOOGLE_CLOUD_PROJECT environment variable. Skipping integration test.")
+		}
+	}
+	apiCtx := context.Background()
+	profilerAPIClient, err := gcpProfiler.NewService(apiCtx, option.WithScopes(gcpProfiler.CloudPlatformScope))
+	if err != nil {
+		t.Fatalf("Failed to create Cloud Profiler API client: %v", err)
+	}
+	profilesFoundForTestServiceAndVersion, err := listProfilesForServiceAndVersion(apiCtx, t, profilerAPIClient, projectID)
+	if err != nil {
+		t.Fatalf("listProfilesForServiceAndVersion: %v", err)
+	}
+
+	// Validate the result.
+	expectedProfileTypesInAPI := map[string]bool{
+		"CPU":        false,
+		"HEAP":       false,
+		"THREADS":    false,
+		"CONTENTION": false,
+		"HEAP_ALLOC": false,
+	}
+	for _, p := range profilesFoundForTestServiceAndVersion {
+		if p.ProfileType != "" {
+			if _, ok := expectedProfileTypesInAPI[p.ProfileType]; ok {
+				expectedProfileTypesInAPI[p.ProfileType] = true
+				t.Logf("Marked API profile type '%s' as found.", p.ProfileType)
+			}
+		}
+	}
+	atleastOneFound := false
+	allExpectedFound := true
+	for typeStr, found := range expectedProfileTypesInAPI {
+		if !found {
+			allExpectedFound = false
+			t.Logf("Expected profile type '%s' was NOT found.", typeStr)
+		} else {
+			atleastOneFound = true
+			t.Logf("Found profile type '%s'.", typeStr)
+		}
+	}
+	if !allExpectedFound {
+		t.Logf("Total profiles found for this service/version during test: %d", len(profilesFoundForTestServiceAndVersion))
+		if len(profilesFoundForTestServiceAndVersion) > 0 {
+			t.Log("Details of profiles found:")
+			for _, p := range profilesFoundForTestServiceAndVersion {
+				t.Logf("  - Name: %s, Type: %s, Labels: %v", p.Name, p.ProfileType, p.Deployment.Labels)
+			}
+		}
+	} else {
+		t.Log("All expected profile types (CPU, HEAP) were successfully found for the test service and version.")
+	}
+	if !atleastOneFound {
+		t.Errorf("Failed: none of the profile found.")
+	}
```

Comment:
Consider something like this:
```suggestion
	for typeStr, found := range expectedProfileTypesInAPI {
		if !found {
		  <message saying that not all profile types were found and print the map>
		}
	}
```